diagnostics/collect-wsl-logs.ps1

```diff
@@ -131,86 +131,56 @@ if ($Dump)
+    Write-Output "Get-NetFirewallHyperVProfile -PolicyStore ActiveStore" | Out-File -FilePath "$folder/Get-NetFirewallHyperVProfile.log" -Append
+    Get-NetFirewallHyperVProfile -PolicyStore ActiveStore | Out-File -FilePath "$folder/Get-NetFirewallHyperVProfile.log" -Append
 
-    Write-Output "Get-NetFirewallHyperVVMCreator"  | Out-File -FilePath $wslNetworkLog -Append
-    Get-NetFirewallHyperVVMCreator  | Out-File -FilePath $wslNetworkLog -Append
+    Write-Output "Get-NetFirewallHyperVPort" | Out-File -FilePath "$folder/Get-NetFirewallHyperVPort.log" -Append
+    Get-NetFirewallHyperVPort | Out-File -FilePath "$folder/Get-NetFirewallHyperVPort.log" -Append
 
-    Write-Output -----------------------------------------------------------------------   | Out-File -FilePath $wslNetworkLog -Append
-
-    Write-Output "Get-NetFirewallHyperVVMSetting -PolicyStore ActiveStore"  | Out-File -FilePath $wslNetworkLog -Append
-    Get-NetFirewallHyperVVMSetting -PolicyStore ActiveStore  | Out-File -FilePath $wslNetworkLog -Append
-
-    Write-Output -----------------------------------------------------------------------   | Out-File -FilePath $wslNetworkLog -Append
-
-    Write-Output "Get-NetFirewallHyperVProfile -PolicyStore ActiveStore"  | Out-File -FilePath $wslNetworkLog -Append
-    Get-NetFirewallHyperVProfile -PolicyStore ActiveStore  | Out-File -FilePath $wslNetworkLog -Append
-
-    Write-Output -----------------------------------------------------------------------   | Out-File -FilePath $wslNetworkLog -Append
-
-    $hypervVports = Get-NetFirewallHyperVPort
-    if (!$hypervVports)
-    {
-        Write-Output "Get-NetFirewallHyperVPort returned no ports"  | Out-File -FilePath $wslNetworkLog -Append
-    }
-    else
-    {
-        Write-Output "Get-NetFirewallHyperVPort"  | Out-File -FilePath $wslNetworkLog -Append
-        Get-NetFirewallHyperVPort  | Out-File -FilePath $wslNetworkLog -Append
-    }
-
-    Write-Output -----------------------------------------------------------------------   | Out-File -FilePath $wslNetworkLog -Append
-
-    Write-Output "hnsdiag.exe list all" | Out-File -FilePath $wslNetworkLog -Append
+    Write-Output "hnsdiag.exe list all" | Out-File -FilePath "$folder/hnsdiag_list_all.log" -Append
     Start-Process -FilePath "hnsdiag.exe" -ArgumentList "list all" -RedirectStandardOutput $stdOutLog -RedirectStandardError $stdErrLog -NoNewWindow -wait
-    Get-Content $stdOutLog, $stdErrLog | Out-File -FilePath $wslNetworkLog -Append
-
-    Write-Output -----------------------------------------------------------------------   | Out-File -FilePath $wslNetworkLog -Append
+    Get-Content $stdOutLog, $stdErrLog | Out-File -FilePath "$folder/hnsdiag_list_all.log" -Append
 
-    Write-Output "hnsdiag.exe list endpoints -df" | Out-File -FilePath $wslNetworkLog -Append
+    Write-Output "hnsdiag.exe list endpoints -df" | Out-File -FilePath "$folder/hnsdiag_list_endpoints.log" -Append
```

Comment:
Could we simplify by : `& hnsdiag.exe list endpoints -df  2>&1 >> "$folder/hnsdiag_list_endpoints.log"`  ?

Reply:
(Same for the below calls)

Reply:
thanks :) this should help remove the need for the additional stderr and stdout files

Reply:
ended up using the "$cmd | cmd | Out-Null" construct to run cmd commands